From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Mina Her <minacle@live.com>
Date: Tue, 27 May 2025 16:48:13 +0900
Subject: [PATCH 1/2] Add Git repository information to Dockerfile


diff --git a/Dockerfile b/Dockerfile
index 6cb4a2a1c0..92aa459da2 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -30,6 +30,12 @@ FROM ${BASE_REGISTRY}/ruby:${RUBY_VERSION}-slim-${DEBIAN_VERSION} AS ruby
 ARG MASTODON_VERSION_PRERELEASE=""
 # Append build metadata or fork information to version.rb [--build-arg MASTODON_VERSION_METADATA="pr-123456"]
 ARG MASTODON_VERSION_METADATA=""
+# GitHub repository to use for 'View source code' link in Mastodon footer [--build-arg GITHUB_REPOSITORY="mastodon/mastodon"]
+ARG GITHUB_REPOSITORY=""
+# Git repository URL to use for 'View source code' link in Mastodon footer [--build-arg SOURCE_BASE_URL="https://github.com/$GITHUB_REPOSITORY"]
+ARG SOURCE_BASE_URL=""
+# Precise tag or branch to use for 'View source code' link in Mastodon footer [--build-arg SOURCE_TAG="v4.3.2"]
+ARG SOURCE_TAG=""
 # Will be available as Mastodon::Version.source_commit
 ARG SOURCE_COMMIT=""
 
@@ -48,9 +54,12 @@ ARG GID="991"
 
 # Apply Mastodon build options based on options above
 ENV \
-  # Apply Mastodon version information
+  # Apply Mastodon version and Git repository information
   MASTODON_VERSION_PRERELEASE="${MASTODON_VERSION_PRERELEASE}" \
   MASTODON_VERSION_METADATA="${MASTODON_VERSION_METADATA}" \
+  GITHUB_REPOSITORY="${GITHUB_REPOSITORY}" \
+  SOURCE_BASE_URL="${SOURCE_BASE_URL}" \
+  SOURCE_TAG="${SOURCE_TAG}" \
   SOURCE_COMMIT="${SOURCE_COMMIT}" \
   # Apply Mastodon static files and YJIT options
   RAILS_SERVE_STATIC_FILES=${RAILS_SERVE_STATIC_FILES} \

From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Mina Her <minacle@live.com>
Date: Wed, 28 May 2025 04:10:29 +0000
Subject: [PATCH 2/2] Use presence to determine whether to use the default


diff --git a/config/mastodon.yml b/config/mastodon.yml
index 4585e1f2ae..2cdcc85152 100644
--- a/config/mastodon.yml
+++ b/config/mastodon.yml
@@ -6,7 +6,7 @@ shared:
   software_update_url: <%= ENV.fetch('UPDATE_CHECK_URL', 'https://api.joinmastodon.org/update-check')&.to_json %>
   source:
     base_url: <%= ENV.fetch('SOURCE_BASE_URL', nil)&.to_json %>
-    repository: <%= ENV.fetch('GITHUB_REPOSITORY', 'mastodon/mastodon') %>
+    repository: <%= ENV.fetch('GITHUB_REPOSITORY', nil).presence || 'mastodon/mastodon' %>
     tag: <%= ENV.fetch('SOURCE_TAG', nil) %>
   version:
     metadata: <%= ENV.fetch('MASTODON_VERSION_METADATA', nil)&.to_json %>
